addons/controllers/clusterbootstrap_controller.go

```diff
@@ -1285,10 +1286,16 @@ func (r *ClusterBootstrapReconciler) removeFinalizersFromClusterResources(cluste
 	return nil
 }
 
-func (r *ClusterBootstrapReconciler) reconcileDelete(cluster *clusterapiv1beta1.Cluster, remoteClient client.Client, log logr.Logger) (ctrl.Result, error) {
+func (r *ClusterBootstrapReconciler) reconcileDelete(cluster *clusterapiv1beta1.Cluster, log logr.Logger) (ctrl.Result, error) {
 	okToRemoveFinalizers := false
 	var err error
 
+	remoteClient, err := util.GetClusterClient(r.context, r.Client, r.Scheme, clusterapiutil.ObjectKey(cluster))
+	if err != nil {
+		log.Error(err, "Error getting remote cluster client")
```

Comment:
suggestion: Don't log errors unless you're adding extra debug info. Instead, wrap it and return it like:

```go
return ctrl.Result{Requeue: true}, fmt.Errorf("failed to get remote cluster client: %w", err)
```